src/qengine/opencl.cpp

```diff
@@ -920,13 +912,6 @@ void QEngineOCL::ProbRegAll(const bitLenInt& start, const bitLenInt& length, rea
 // Returns probability of permutation of the register
 real1 QEngineOCL::ProbMask(const bitCapInt& mask, const bitCapInt& permutation)
 {
-    // We might have async execution of gates still happening.
-    clFinish();
```

Comment:
Double check: clFinish not needed here?

Reply:
clFinish() sits at the top of NormalizeState.

clFinish() was called specifically in preparation for NormalizeState, generally. This was because, the normalization constant might be waiting for an asynchronous kernel. However, I re-flowed everything so NormalizeState is called regardless of the immediate value of the normalization constant, then clFinish() is called at the method's top, and NormalizeState exits early after the block, if runningNorm is definitely equal to 1.

No performance hit; safer; simpler.